components/buttons/Button.tsx

```diff
@@ -0,0 +1,77 @@
+import { twMerge } from "tailwind-merge";
+import Link from 'next/link'
+
+type IButtonProps = {
+  text: string;
+  icon?: React.ReactNode;
+  iconPosition?: 'left' | 'right';
+  target?: string;
+  bgClassName?: string;
+  textClassName?: string;
+  buttonSize?: 'small' | 'default';
+  type?: 'submit' | 'reset' | 'button';
+} & 
+({
+  href: string;
+} & React.AnchorHTMLAttributes<HTMLAnchorElement> | {
+  href: undefined | null;
+} & React.ButtonHTMLAttributes<HTMLButtonElement>)
+
+
+/**
+ * @description The Button component is a reusable button component that can be used to render a button or an anchor tag. It accepts a variety of props to customize the button's appearance and behavior.
+ * @description The component accepts button or anchor tag props based on the presence of the href prop. If the href prop is present, the component will render an anchor tag, otherwise it will render a button tag.
+ */
```

Comment:
I think you have to provide jsdoc comment according to the props pass inside the component.

Reply:
Sure I will add the JSDoc Comments.